process/executionTrack/executionResultsTracker_test.go

```diff
@@ -15,112 +16,160 @@ func TestNewExecutionResultsTracker(t *testing.T) {
+	require.NoError(t, err)
+	require.Equal(t, execResult, lastNotarizedResult)
 }
 
-func TestGetPendingExecutionResults(t *testing.T) {
+func TestAddExecutionResult_AllBranches(t *testing.T) {
 	t.Parallel()
 
 	tracker, _ := NewExecutionResultsTracker()
-	tracker.lastNotarizedResult = &block.ExecutionResult{Nonce: 9}
 
-	res1 := &block.ExecutionResult{HeaderHash: []byte("h1"), Nonce: 10}
-	res2 := &block.ExecutionResult{HeaderHash: []byte("h2"), Nonce: 11}
-	_ = tracker.AddExecutionResult(res1)
-	_ = tracker.AddExecutionResult(res2)
+	// 1. Nil execution result
+	err := tracker.AddExecutionResult(nil)
+	require.ErrorIs(t, err, ErrNilExecutionResult)
 
-	results, err := tracker.GetPendingExecutionResults()
-	require.NoError(t, err)
-	require.Len(t, results, 2)
-	require.Same(t, res1, results[0])
-	require.Same(t, res2, results[1])
+	// 2. Nil lastNotarizedResult
+	execResult := &block.ExecutionResult{HeaderHash: []byte("hash1"), Nonce: 1}
+	err = tracker.AddExecutionResult(execResult)
+	require.ErrorIs(t, err, ErrNilLastNotarizedExecutionResult)
 
-	tracker, _ = NewExecutionResultsTracker()
-	tracker.lastNotarizedResult = &block.ExecutionResult{Nonce: 9}
-	_ = tracker.AddExecutionResult(&block.ExecutionResult{HeaderHash: []byte("h1"), Nonce: 10})
-	_ = tracker.AddExecutionResult(&block.ExecutionResult{HeaderHash: []byte("h3"), Nonce: 12}) // skip 11
+	// Set lastNotarizedResult for further tests
+	tracker.lastNotarizedResult = &block.ExecutionResult{Nonce: 10}
 
-	results, err = tracker.GetPendingExecutionResults()
+	// 3. executionResult.Nonce < lastNotarizedResult.Nonce
+	execResult = &block.ExecutionResult{HeaderHash: []byte("hash2"), Nonce: 9}
+	err = tracker.AddExecutionResult(execResult)
 	require.Error(t, err)
-	require.Nil(t, results)
-}
-
-func TestGetPendingExecutionResultsByHash(t *testing.T) {
-	t.Parallel()
+	require.Contains(t, err.Error(), "is lower than last notarized nonce")
 
-	tracker, _ := NewExecutionResultsTracker()
-	tracker.lastNotarizedResult = &block.ExecutionResult{Nonce: 9}
+	// 4. lastExecutedResults.Nonce != executionResult.Nonce-1
+	execResult = &block.ExecutionResult{HeaderHash: []byte("hash3"), Nonce: 12}
+	err = tracker.AddExecutionResult(execResult)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "is lower than last executed nonce")
 
-	res1 := &block.ExecutionResult{HeaderHash: []byte("h1"), Nonce: 10}
-	_ = tracker.AddExecutionResult(res1)
+	// 6. Success path
+	execResult = &block.ExecutionResult{HeaderHash: []byte("hash3"), Nonce: 11}
+	err = tracker.AddExecutionResult(execResult)
+	require.Nil(t, err)
 
-	res, err := tracker.GetExecutionResultByHash([]byte("h1"))
-	require.NoError(t, err)
-	require.Equal(t, res1, res)
+	execResult = &block.ExecutionResult{HeaderHash: []byte("hash3"), Nonce: 13}
+	tracker.lastExecutedResultHash = []byte("h1")
+	err = tracker.AddExecutionResult(execResult)
+	require.Contains(t, err.Error(), "last executed result not found")
```

Comment:
this can panic.. perhaps require.Error(t, err) before

Reply:
added